src/raster/array.jl

```diff
@@ -1,89 +1,106 @@
+import DiskArrays: eachchunk, GridChunks
+import DiskArrays
 const AllowedXY = Union{Integer,Colon,AbstractRange}
 const AllowedBand = Union{Integer,Colon,AbstractArray}
 
 
+# Define a RasterDataset type
+struct RasterDataset{T,DS} <: AbstractDiskArray{T,3}
+    ds::DS
+    size::Tuple{Int,Int,Int}
+end
+#Forward a few functions
```

Comment:
Do we have any general principles for the set of functions we want to be defining on a `RasterDataset`?

Reply:
Basically everything that is relevant for raster-like datasets. Is there a way to get a comprehensive list of everything which is defined for AbstractDataset, so one can decide globally? Of course I would omit things like getfeature etc...

Reply:
Ah thanks for explaining -- that wasn't obvious to me because `DS` isn't constrained to be `<: AbstractDataset`.

Reply:
I don't yet know how to get such a list of methods either.. Possibly `methodswith()`?